pkg/platform/runtime/runtime.go

```diff
@@ -127,14 +131,7 @@ func (r *Runtime) Env(inherit bool, useExecutors bool) (map[string]string, error
 
 	envDef, err := r.envDef()
 	if !r.envAccessed {
-		if err != nil {
-			r.analytics.EventWithLabel(anaConsts.CatRuntime, anaConsts.ActRuntimeFailure, anaConsts.LblRtFailEnv)
-		} else {
-			r.analytics.Event(anaConsts.CatRuntime, anaConsts.ActRuntimeSuccess)
-			if r.target.Trigger().IndicatesUsage() {
-				r.recordUsage()
-			}
-		}
+		r.recordCompletion(err)
 		r.envAccessed = true
 	}
```

Comment:
Since the only thing interesting being protected here is to call r.recordCompletion, the r.completed guard should be enough and r.envAccessed can be dropped.